src/menu/menu.c

```diff
@@ -843,22 +920,89 @@ menu_open(struct menu *menu, int x, int y)
 	menu->server->input_mode = LAB_INPUT_STATE_MENU;
 }
 
+static void
+parse_pipemenu(struct menuitem *item)
+{
+	FILE *fp = popen(item->execute, "r");
+	if (!fp) {
+		return;
+	}
+	struct server *server = item->parent->server;
+
+	/*
+	 * Pipemenus do not contain a toplevel <menu> element so we have to
+	 * create that first `struct menu`.
+	 *
+	 * We just use for the 'execute' command as a menu id. If this creates a
+	 * name-space issue, we'll have to uniquify them.
+	 */
+	++menu_level;
+	current_menu = menu_create(server, item->execute, NULL);
+	current_menu->is_pipemenu = true;
+	wlr_log(WLR_INFO, "parse pipemenu '%s'", item->execute);
+	parse(server, fp);
+	current_menu = current_menu->parent;
+	item->submenu = menu_get_by_id(server, item->execute);
+	--menu_level;
+
+	pclose(fp);
+}
+
 static void
 menu_process_item_selection(struct menuitem *item)
 {
 	assert(item);
 
+	/* Do not keep selecting the same item */
+	static struct menuitem *last;
+	if (item == last) {
+		return;
+	}
+	last = item;
```

Comment:
What if we select an item, close the menu, open the menu and try to select the same item again here?
Wouldn't the static `last` cause an early return in that situation?

Reply:
No. Don't think so.